runtime/parachains/src/dmp.rs

```diff
@@ -228,7 +236,11 @@ impl<T: Config> Pallet<T> {
 			} else {
 				*q = q.split_off(processed_downward_messages);
 			}
+			q.len()
 		});
+		if q_len <= MAX_MESSAGE_QUEUE_SIZE {
+			Self::decrement_fee_factor();
```

Comment:
Are we decrementing once per batch of messages sent but incrementing for each message? If so the fee will stay high...

Reply:
Yes, the idea is that the fee would rise quickly upwards, and would only slowly lower each block, and only when it is not overqueued.